### PR TITLE
[PT FE] Add aten::is_nonzero

### LINKED ISSUE #20582: [Good First Issue]: Support aten::is_nonzero for pytorch models
### Context

OpenVINO component responsible for support of PyTorch models is called as PyTorch Frontend (PT FE). PT FE converts a model represented as TorchScript model to a model in [OpenVINO opset](https://docs.openvino.ai/latest/openvino_docs_ops_opset.html).

### What needs to be done?

- Implement conversion rule and/or transformation to support the new operation.
- Implement operation tests in tests/layer_tests/pytorch_tests. Please consider different data types, but keep reasonable number of test cases

### Example Pull Requests

#18998

### Resources

- [torch.is_nonzero description](https://pytorch.org/docs/stable/generated/torch.is_nonzero.html)
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [OpenVINO PyTorch Frontend](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/pytorch/README.md)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)

### Contact points

@openvinotoolkit/openvino-pytorch-frontend-maintainers 

### Ticket

TBD

### PR DESCRIPTION
**Pull request by siddhant-0707**
### Details:
 - Add support for aten::is_nonzero for pytorch models

### Tickets:
 - Closes #20582

### CODE REVIEW

**Comment by siddhant-0707** [comment]
Hey @mvafin could you take a look.

**Review by p-wysocki — COMMENTED** [review]
Thank you for your contribution, you're making a lot of great PRs!

The tests are failing due to some argument type mismatch:
```bash
E       RuntimeError: Check 'element::Type::merge(result_et, node->get_input_element_type(0), node->get_input_element_type(1))' failed at src/core/src/op/util/elementwise_args.cpp:25:
E       While validating node 'opset1::NotEqual aten::is_nonzero/NotEqual_2 (input_tensor[0]:i64[?], aten::is_nonzero/Constant_1[0]:boolean[1]) -> (boolean[...])' with friendly_name 'aten::is_nonzero/NotEqual_2':
E       Arguments do not have the same element type (arg0 element type: i64, arg1 element type: boolean).
```

File: src/frontends/pytorch/src/op/is_nonzero.cpp
Comment (mvafin) [inline src/frontends/pytorch/src/op/is_nonzero.cpp:33]: Since pytorch will throw exception when tensor is not a single we don't have to check that case. We will also throw on if shape will be unexpected.

File: src/frontends/pytorch/src/op/is_nonzero.cpp
Comment (mvafin) [inline src/frontends/pytorch/src/op/is_nonzero.cpp:43]: You can use `align_eltwise_input_types` helper to align types of 0 tensor and input and

**Comment by siddhant-0707** [comment]
Should I remove `np.array([1, 3, 5])` from `input_tensor` in the test?

**Comment by mvafin** [comment]
> Should I remove `np.array([1, 3, 5])` from `input_tensor` in the test? It causes the RuntimeError as expected

Yes, please remove it

File: src/frontends/pytorch/src/op/is_nonzero.cpp
Comment (mvafin) [inline src/frontends/pytorch/src/op/is_nonzero.cpp:28]: I don't think there is a need for special case for bool, `align_eltwise_input_types` should work for that case too. You can create 0 of lowest type, which is `boolean` and it will be upgraded by `align_eltwise_input_types` to appropriate type - float or int.

**Review by mvafin — APPROVED** [review]
LGTM, lets wait for jenkins results before merge

**Comment by siddhant-0707** [comment]
build_jenkins

**Comment by mvafin** [comment]
build_jenkins

**Comment by rkazants** [comment]
@mvafin, please advice how to fix the current issue in the tests:

```
    def convert(self, model: Union[Model, InputModel]) -> Model:
>       converted_model = super().convert(model)
E       openvino._pyopenvino.OpConversionFailure: Check 'is_conversion_successful' failed at src/frontends/pytorch/src/frontend.cpp:140:
E       FrontEnd API failed with OpConversionFailure:
E       Model wasn't fully converted. Failed operations detailed log:
E       -- ov::align_types with a message:
E       This is internal operation for type alignment and should be removed at normalization step. It can't be removed if types can't be resolved.
E       Summary:
E       -- Conversion is failed for: ov::align_types
```

File: src/frontends/pytorch/src/op/is_nonzero.cpp
Comment (mvafin) [inline src/frontends/pytorch/src/op/is_nonzero.cpp:24]: Please use `ConvertLike` instead of `align_eltwise_input_types`. It is not required here.

**Comment by mvafin** [comment]
build_jenkins

**Review by mvafin — APPROVED** [review]

**Comment by rkazants** [comment]
build_jenkins

**Pull request merged**
